src/pim-types.ts

```diff
@@ -171,6 +171,14 @@ export class TaskType extends EventType {
     return !!((this.startDate?.isDate) || (this.dueDate?.isDate));
   }
 
+  set tags(tagHash: any) {
+    this.component.updatePropertyWithValue('tags', JSON.stringify(tagHash));
+  }
+
+  get tags() {
+    return JSON.parse(this.component.getFirstPropertyValue('tags'));
```

Comment:
I'll properly review everything slightly later, when you make more progress, but one important comment: this shouldn't be called "tags" but "properties", and you shouldn't JSON.stringfiy it, but rather it should be an array of strings.

This is not something that comes from me, it comes from the [ical standard](https://tools.ietf.org/html/rfc5545#section-3.8.1.2). This object that you're updating is an ical object that then needs to be serialized to the ical format. It doesn't change much, just make sure to rename it and change the value. Additional properties of tags, such as colour, should be saved elsewhere, and I'll happily explain it when you get to saving those. Though those should be maintained at a per-journal capacity as part of the journal metadata.

Reply:
I see. I think I've got it now. It was a little confusing to figure out how to do multi value properties, but looks like I need to explicitly add the "CATEGORIES" property to the `TaskType`.

Reply:
> Additional properties of tags, such as colour, should be saved elsewhere, and I'll happily explain it when you get to saving those. Though those should be maintained at a per-journal capacity as part of the journal metadata.

Could I also save a list of all the tags this way? If so, how would I go about that?

Reply:
The list of tags is external to specific tasks. It should probably be stored in each journal's metadata. I'm happy to explain to you how to do it (it would require a tiny bit of code on my side), though I think it's too early for that, just save it locally (and globally) for now, or heck, make the tags hardcoded while you are working on it, and ping me again in a few days / a week and I'll add the ability to set generic journal metada.